### ISSUE TITLE
OOps/pffft.c doesn't use/deal with the HAVE_NEON build flag

### ISSUE THREAD

**Issue by nullr0ute** [csound/csound#933]
Some ARMv7 SoCs don't have NEON SIMD, two examples are tegra2 and the OLPC XO-1.75. As a result a lot of generic distros build without NEON optimisations and packages can then use fast paths when they detect SIMD is available at run time.

The OOps/pffft.c doesn't take notice of the HAVE_NEON cmake flag so specifying -DHAVE_NEON:BOOL=OFF will still result in build failures in this scenario. Instead to get it to build you have to also add the a "sed -i 's*//#define PFFFT_SIMD_DISABLE*#define PFFFT_SIMD_DISABLE*' OOps/pffft.c" hack to make it build.

The OOps/pffft.c file should be updated so that it'll take notice of the HAVE_NEON flag.

**Comment by vlazzarini** [comment]
Can't you you set -DPFFFT_SIMD_DISABLE in cmake instead of -DHAVE_NEON:BOOL=OFF?
I am happy to make changes but I want to understand the issue first. I wonder how debian packaging deals with this (@fsateler?).

**Comment by fsateler** [comment]
> I will investigate.

If someone has a simple CSD that would trigger the pffft neon paths for testing that would be great.

**Comment by vlazzarini** [comment]
```
<CsoundSynthesizer>
<CsOptions>
--fftlib=1 
</CsOptions>
<CsInstruments>
instr 1
asig oscili 0dbfs/2, A4
fsig pvsanal asig, 1024,256,1024,1
asig pvsynth fsig
out asig

endin
</CsInstruments>
<CsScore>
i1 0 10
</CsScore>
</CsoundSynthesizer>
```

**Comment by vlazzarini** [comment]
Any news on this one? @fsateler

**Comment by fsateler** [comment]
Sorry, I lost track of this. I'll check during the weekend

**Comment by vlazzarini** [comment]
Thanks!

**Comment by fsateler** [comment]
The csd does appear to work on the armel porterbox. This box does not appear to have NEON:

`Features	: half thumb fastmult vfp edsp thumbee vfpv3 tls idiva idivt vfpd32 lpae`

**Comment by nullr0ute** [comment]
NEON isn't a requirement until ARMv8. For < ARMv8, which these days is mostly ARMv7 with the most common < v7 being the Raspberry Pi 0/1 devices (rpi2 and later are ARMv7 or v8). Ideally on ARMv7 there should be run time detection/optimisation for NEON, for example Fedora compile their binaries not to require NEON but a lot of libraries if it's detected at runtime will have fast paths for NEON optimisation.

**Comment by jpffitch** [comment]
Is thix still unresolved?

**Comment by kunstmusik** [comment]
@vlazzarini @fsateler is this issue still relevant or can this be closed? If not, I will schedule for 7.0.0.

**Comment by vlazzarini** [comment]
This can be closed now. I've added what @fsateler suggested.
